sklearn/manifold/_isomap.py

```diff
@@ -166,6 +172,22 @@ def _fit_transform(self, X):
         G = self.dist_matrix_ ** 2
         G *= -0.5
 
+        kpca_eigen_solver = self.eigen_solver
+        if kpca_eigen_solver == 'auto':
+            # use the legacy 'auto' to avoid the randomized method as it
+            # does not skip the large negative eigenvalues for now
```

Comment:
Note: this calls for a later TODO (in a distinct issue/pull request): the reason why I was not able to make Isomap benefit from the acceleration is that the randomized svd method returns the largest eigenvalues in magnitude, not in sign. So if you ask the randomized svd for 2 eigenvals (as isomap is asking) and your matrix is not PSD (as often in isomap) you may have 1 large positive and 1 large negative, or 2 large negative eigenvalues. There might be a clever way to adapt the method, for example by asking for more (10? 20?) and only keep the largest 2 with positive sign, but this is non-trivial and also it would require to pass to kPCA a dedicated flag indicating to skip PSD enforcement.

All in all, a great thing to do for all isomap users, but... in a separate issue/PR. What do you think ?

Reply:
But then shouldn't this caveat be documented under the docstring for `kpca_eigen_solver == 'randomized'`?

Reply:
done